examples/purchase.rs

```diff
@@ -53,35 +56,36 @@ async fn main() {
 
     let bot = Bot::from_env().auto_send();
 
+    Dispatcher::builder(bot, schema())
+        .dependencies(dptree::deps![InMemStorage::<State>::new()])
+        .build()
+        .setup_ctrlc_handler()
+        .dispatch()
+        .await;
+}
+
+fn schema() -> UpdateHandler<Box<dyn std::error::Error + Send + Sync + 'static>> {
```

Comment:
What's the point in moving this to a separate function?

Reply:
Separate the startup procedure (dispatcher and bot creation) from a dispatch schema. With this function, a reader can focus on either of these two responsibilities without distraction.

Reply:
Will you do the same for other examples?